### DISCUSSION TITLE
What is the key differences between MimeKit and MimeKitLite?

### DISCUSSION THREAD

**Discussion by ajtruckle** [jstedfast/MimeKit#1023] [General]
I am using `MimeKit `but noticed the Lite version in the NuGet list.

**Comment by ajtruckle** [comment]
Just found:

https://stackoverflow.com/a/75768384/2287576

> `MimeKitLite `contains a subset of the `MimeKit `API that does not include PGP, S/MIME or DKIM/ARC support. In other words, most of the `MimeKit.Cryptography` namespace has been ripped out in `MimeKitLite`.

**Comment by ajtruckle** [comment]
It is possible to list the parent assemblies or classes? So I can identify if I am using them or not?

**Comment by jstedfast** [comment]
If you are using anything in the MimeKit.Cryptography namespace, there is a high probability that you are using code that is stripped out in MimeKitLite.

My advice is that if you are writing a *library* meant to be used by other projects, use MimeKit. If you are writing an *application* and you don't care about cryptography, use MimeKitLite.

**Comment by ajtruckle** [comment]
I am writing an application. The class that specifically uses `MimeKit` has these headers:

```
using Google.Apis.Auth.OAuth2;
using Google.Apis.Gmail.v1;
using Google.Apis.Gmail.v1.Data;
using Google.Apis.Services;
using Google.Apis.Util.Store;
using MimeKit;
using NLog;
```

So it looks like it might be possible to add `MimeKitLite` and remove `MimeKit`.

**Reply by jstedfast** [reply]
FWIW, the *plan* is to drop the need for MimeKitLite in v5.0 by splitting MimeKit itself into MimeKit.Core (which would effectively be the same as MimeKitLite) and MimeKit (or MimeKit.Cryptography?) which would add the MimeKit.Cryptography namespace.

See https://github.com/jstedfast/MimeKit/issues/820

There's already *some* progress towards this, but it has been very difficult to untangle in a way that allows the modular addition.

**Reply by ajtruckle** [reply]
I see! 👌So is it worth leaving it set as `MimeKit` and then change to `MimeKit.Core` at a future date?

**Reply by jstedfast** [reply]
That's up to you. The only thing you really save by moving to MimeKitLite is reducing app size by removing BouncryCastle (which GoogleApis might depend on anyway).

**Comment by ajtruckle** [comment]
@jstedfast  
> That's up to you. The only thing you really save by moving to `MimeKitLite `is reducing app size by removing `BouncryCastle` (which GoogleApis might depend on anyway).

FYI. I just removed the full version and installed the lite version. All is good and `BouncryCastle` is nowhere to be seen. 😊
